File: warmup/manasastones/manasastones.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"strconv"
	"strings"
)

func main() {
	//file := os.Stdin
	file, _ := os.Open(`.\data\manasastones.txt`)
	rawBytes, _ := ioutil.ReadAll(file)

	cases := strings.Fields(string(rawBytes))[1:]

	for i := 0; i <= len(cases)-3; i += 3 {
		numStones, _ := strconv.Atoi(cases[i+0])
		diffA, _ := strconv.Atoi(cases[i+1])
		diffB, _ := strconv.Atoi(cases[i+2])
		lastStone(numStones, diffA, diffB)
	}
}

func lastStone(numStones, diffA, diffB int) {
	if numStones == 1 {
		fmt.Println("0")
		return
	}

	if diffA == diffB {
		fmt.Printf("%v\n", diffA*(numStones-1))
		return
	}

	if diffA > diffB {
		diffA, diffB = diffB, diffA
	}

	last := diffA * (numStones - 1)

	for i := 0; i < numStones; i++ {
		fmt.Printf("%v ", last-i*diffA+i*diffB)
	}

	fmt.Println()
}

File: warmup/manasastones/manasastones.py
#from sys import stdin

file = open( r".\data\manasastones.txt" )
data = list(map(int, file.read().strip().split()))[1:]
file.close()

def lastStone(numStones, diffA, diffB):
	diffs = set()
	for i in range(numStones):
		diffs.add( (numStones-i-1)*diffA + i*diffB)
	return diffs

# For each test case triple...
for i in range(0, len(data), 3):
	print( *sorted( lastStone(*data[i:i+3]) ) )

